promql/engine_test.go

```diff
@@ -3708,3 +3709,76 @@ histogram {{sum:4 count:4 buckets:[2 2]}} {{sum:6 count:6 buckets:[3 3]}} {{sum:
 		},
 	})
 }
+
+func TestRateAnnotations(t *testing.T) {
+	testCases := map[string]struct {
+		data                       string
+		expr                       string
+		expectedWarningAnnotations []string
+		expectedInfoAnnotations    []string
+	}{
+		"annotation when two samples are selected": {
```

Comment:
I think distinguishing between info and warning annotations would be helpful to readers:
```suggestion
		"info annotation when two samples are selected": {
```